client.go

```diff
@@ -96,3 +96,44 @@ func (c Client[Resource]) GetByID(ctx context.Context, id string, options Option
 	}
 	return resource, nil
 }
+
+// Body to update a document in the collection
+type PatchBody struct {
+	// Set replaces the value of the field with specified value. It is possible also
+	// to use with nested fields: e.g. `"a.b": "update"`
+	Set map[string]any `json:"$set,omitempty"`
+	// Unset a particular document value
+	Unset map[string]bool `json:"$unset,omitempty"`
+	// Inc increment a field by a specified value
+	Inc map[string]int `json:"$inc,omitempty"`
+	// Mul multiply the value of a field by a specified number
+	Mul map[string]int `json:"$mul,omitempty"`
+	// CurrentDate sets the value of a field to the current date. The field MUST
+	// be of type Date
+	CurrentDate map[string]any `json:"$currentDate,omitempty"`
+	// Push appends a value to an array field
+	Push map[string]any `json:"$push,omitempty"`
+	// Pull removes a specified value from an array field
+	Pull map[string]any `json:"$pull,omitempty"`
+	// AddToSet appends a specified value to an array field unless the value is
+	// already present
+	AddToSet map[string]any `json:"$addToSet,omitempty"`
+}
+
+// Patch update an element using commands in PatchBody
+func (c Client[Resource]) Patch(ctx context.Context, id string, body PatchBody, options Options) (*Resource, error) {
```

Comment:
This is the PatchById method, isn't it? I'd change the name to be able to support the `Patch` (used as PatchMany) later

Reply:
PatchMany will be PatchMany, or PathBulk. I could change with updateOne/updateById instead of patch. What do you think?

Reply:
Patch is fine to my rather than "Update" (even though in CRUD you say Update and not patch, but still it properly maps to the endpoint used).

PatchMany and PatchBulk are two different things for our crud service; we should map

* Patch -> PATCH /:id
* PatchMany -> PATCH /
* PatchBulk -> PATCH /bulk

I'd use `PatchById` for coherence though as in 


* PatchById -> PATCH /:id
* Patch -> PATCH /
* PatchBulk -> PATCH /bulk


Still Patch it is too generic, and the final api could be

* PatchById -> PATCH /:id
* PatchMany -> PATCH /
* PatchBulk -> PATCH /bulk

Reply:
Do we have any standard to call these actions? 
I will use the name "Mongo like" as UpdateOne and UpdateMany 
Why implement the patchBulk?

Reply:
Because this is a client for the CRUD Service, not any CRUD therefore I'd keep the API as similar to the service as possible

Reply:
Renamed to PatchById. I agree with the third, it's more clear.